### PR TITLE
fix: change logic to get slidesPerView

### PR DESCRIPTION
**Pull request by kang-heewon**
Hello! I am a fan of this library!
I have the same problem as #52
I identified the problem and tried to patch it.

However, I did not want to affect other logic, so I modified it as simple as possible.
I think you would know the meaning of the things I edited!

I hope this code is reflected or bugs are fixed!

A codepen that I simply tested.
https://codesandbox.io/s/blazing-sun-94qtv?file=/index.html

### CODE REVIEW

File: src/keen-slider.js [hunk 1]
```diff
@@ -402,6 +402,9 @@ function KeenSlider(initialContainer, initialOptions = {}) {
   }
 
   function sliderGetSlidesPerView(option) {
+    if (option > length) {
```
Comment (ruijdacd) [inline src/keen-slider.js:405]: You could effectively change this function to return early, if the type of `option` is function and then have a clean return statement at the end

File: src/keen-slider.js [hunk 2]
```diff
@@ -402,6 +402,9 @@ function KeenSlider(initialContainer, initialOptions = {}) {
...
+      length = option
+    }
     return typeof option === 'function'
       ? option()
       : clampValue(option, 1, Math.max(isLoop() ? length - 1 : length, 1))
```
Comment (ruijdacd) [inline src/keen-slider.js:410]:
```suggestion
    if (typeof option === 'function')
      return option()
      
    if (option > length)
      length = option
      
    return clampValue(option, 1, Math.max(isLoop() ? length - 1 : length, 1))
```

**Review by ruijdacd — APPROVED** [review]

File: src/keen-slider.js (hunk 1 above)
Comment (kang-heewon) [reply src/keen-slider.js]: I agree.

**Review by ruijdacd — APPROVED** [review]

**Comment by rcbyr** [comment]
Hello @kang-heewon 

thanks for using the library and your contribution.
Can you give me some information on why you need that "feature" or what you want to achieve. If you have fewer slides then space to display them. Why do you need a slider?

**Comment by ruijdacd** [comment]
@rcbyr That's quite an interesting observation, but I would still assume that the behaviour should be the same, no matter if there are enough slides or not, just for the sake of consistency.

This could perhaps be an option that can be toggled depending on the use case.

**Comment by spacedawwwg** [comment]
> Hello @kang-heewon
> 
> thanks for using the library and your contribution.
> Can you give me some information on why you need that "feature" or what you want to achieve. If you have fewer slides then space to display them. Why do you need a slider?

I a responsive site, the number of slides could fluctuate meaning at certain screen sizes the slider would kick-in (example, on rotation of a phone or tablet)

**Comment by kang-heewon** [comment]
@rcbyr Can you reflect or solve the problem?

**Review by ufon — APPROVED** [review]

**Comment by kang-heewon** [comment]
If the length of the item get from the server is less than the slidesperview, it will not work as expected.
If you don't like the solution I'm suggesting, how about creating a "fixed" (whatever name) option?

**Comment by EduardoLopes** [comment]
> If you have fewer slides then space to display them. Why do you need a slider?
Sometimes the api doesn't return enough items, so there is a empty space that will be filled up i the future, so we need the component to be a slider.

This doesn't need to be the default behavior. The `fixedSlideWidth` option seems to be a good idea.

I really like this library, and this is the only feature the is missing for me.

**Comment by andrey-hohlov** [comment]
Any chance for a merge?

We have two projects in development that wait for this change before release. And existing workaround with !important styles in our cases is bad for us :(

**Comment by rcbyr** [comment]
Sorry, this has taken so long. I added a new property autoAdjustSlidesPerView (default: true). Set it to false and you have the desired behavior.

**Pull request merged**